src/modules/audio/audio.c

```diff
@@ -353,6 +346,7 @@ bool lovrSourceIsLooping(Source* source) {
 }
 
 void lovrSourceSetLooping(Source* source, bool loop) {
+  lovrAssert(loop == false || lovrSoundDataIsStream(source->sound) == false, "Can't loop streams");
```

Comment:
Would it be useful to have a looping audio stream if that means that it constantly plays whatever is pushed to it instead of needing to be :played all the time?  Maybe a special case isn't needed here

Reply:
Hmm. I'm not sure that's great. For one thing, the audio code already has a few places where loops very easily turn into infinite loops. E g, setting looping on a stream right now hangs the mix function. Reusing the looping flag for something else feels like it overcomplicates something which is already complicated.

The second reason for not doing it is that in understanding/experience, you would want to ensure that you have enough data buffered so you have headroom for stutters before you play again. This headroom is very implementation specific (low if you're generating data on another thread, higher if you're receiving data from network), so a generic solution would likely just create a lot of stutter as the source goes from playing to non-playing every frame or so.

Maybe it's just my implementations that are too fragile :P But that's my take on why I think stream sources should only played when asked to. If you don't agree, I can just avoid using the flag myself in my high latency scenarios, so I don't mind implementing your suggestion if you still think it's a good idea!